test/behaviors/t_for_all.clj

```diff
@@ -172,3 +175,87 @@
   (for-all
     [x gen/int]
     x => integer?))
+
+(gen-let [s (gen/return "s")]
+  s => "s")
+
+(fact "you can put gen-let inside of a fact"
+  (gen-let [s (gen/return "s")]
+    s => "s")
+  )
+(fact "gen-let composes generators"
+  (gen-let [i (gen/elements [1 2 3])
+            s (gen/return (str i))]
+    (Integer/parseInt s) => i))
+
+(fact "gen-let composes multiple generators"
+  (gen-let [i  gen/s-pos-int
+            is (gen/vector gen/int i)
+            e  (gen/elements is)]
+    (count is) => i
+    (some #{e} is) => e))
+
+(fact ":let syntax allows for simple functions over generated elements"
+  (gen-let [i gen/pos-int
+            :let [s  (str i)
+                  s2 (str s s)]]
+    (Integer/parseInt s2) => (+ (* 10 i) i)))
+
+(fact "a :let block can be in the middle of gen-let bindings"
+  (gen-let [i  (gen/choose 0 9)
+            :let [s  (str i)
+                  s2 (str s s)]
+            xs (gen/elements [s s2])]
+    (->> xs seq (map #(Character/getNumericValue %)) (some #{i})) => i))
+
+(fact "a :let block can be in the beginning of gen-let bindings"
+  (gen-let [:let [i 2
+                  j (+ i 2)]
+            x (gen/vector gen/int i j)]
+    (some #{2 3 4} (count x)) => some?))
+
+(facts "verifying gen-let macro code generation"
+  (fact
+    (parser/roll-up-bindings
```

Comment:
I don't think tests for internal functions should be in the `behaviours` namespaces

Reply:
Makes sense. I don't know where they should be, though. Any suggestion?